src/types/types-external.ts

```diff
@@ -251,6 +251,11 @@ export interface IProduceWithPatches {
 	): Promise<PatchesTuple<Base>>
 }
 
+/**
+ * The type for `recipe function`
+ */
+export type IProducer<T> = (draft: Draft<T>) => ValidRecipeReturnType<Draft<T>> | Promise<ValidRecipeReturnType<Draft<T>>>
```

Comment:
I think it should likely be called `Producer` to stay consistent with `Draft`, `Patch`, `Immutable`, etc.